crates/libmoonlight/src/util.rs

```diff
@@ -1,32 +1,16 @@
-use crate::types::{DetectedInstall, InstallInfo, InstallerResult};
-
-use super::types::{Branch, MoonlightBranch};
+use crate::types::{DetectedInstall, InstallInfo, InstallerResult, Branch};
 use std::path::{Path, PathBuf};
 
 const DOWNLOAD_DIR: &str = "dist";
 pub const PATCHED_ASAR: &str = "_app.asar";
 
-// Reimplementation of electron's appdata stuff
-// Don't like to unwrap here but also if this is broken the entire app is broken
 pub fn get_moonlight_dir() -> PathBuf {
-    let dir = match std::env::consts::OS {
-        "windows" => {
-            let appdata = std::env::var("APPDATA").unwrap();
-            PathBuf::from(appdata).join("moonlight-mod")
-        }
-        "macos" => {
-            let home = std::env::var("HOME").unwrap();
-            PathBuf::from(home).join("Library/Application Support/moonlight-mod")
-        }
-        "linux" => {
-            let home = std::env::var("HOME").unwrap();
-            PathBuf::from(home).join(".config/moonlight-mod")
-        }
-        _ => unimplemented!("Unsupported OS"),
-    };
+    let dir = std::env::var_os("MOONLIGHT_DIR").map(PathBuf::from)
```

Comment:
We shouldn't rely on an external crate to do this - we need to ensure that the paths we select are the same as Electron's, since that's what the client expects. I am fine with an override MOONLIGHT_DIR though - handy for testing.

Reply:
should have parity but oki